server/src/main/java/org/opensearch/cluster/metadata/View.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ConstructingObjectParser;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+/** View of data in OpenSearch indices */
+@ExperimentalApi
+public class View extends AbstractDiffable<View> implements ToXContentObject {
```

Comment:
What are the main differences between a "View" and a "[Filtered Alias](https://opensearch.org/docs/latest/im-plugin/index-alias/#create-filtered-aliases)"? While Filtered Alias may not currently support specifying fields, would that be something that can be added to filter aliases to support the Views use case?

Reply:
Views divorce the existing index concept completely.  An example of this could be augmented view targets with plugins to represent non-index data sources which doesn't fit the existing alias model.

While technical many of the pieces already exist (see also search templates is kind of the other half of this equation), this is a conceptional user model break - what do you think of that?

Reply:
If my understanding is correct, Views are one level above aliases in the sense that they can have 1 to many relationships with indexes, however Aliases have many to 1 relationship with an index. Maybe I'm wrong here, feel free to correct me if so.

Reply:
A view is something (anything?) that can turn a `SearchRequest` into a `SearchResponse`, right? 

(For now, the possible view implementation is one or more index patterns, but later could be other things.)

Reply:
I've written up a blurb for our documentation website that I've got out in draft, I think this might have more of the high-level association - does that work?

- https://github.com/opensearch-project/documentation-website/pull/6394/files?short_path=1f61215#diff-1f61215f0eaaee877d5c392b6ed41d416508d17135db6645fe86d3650c0e912e